src/features/courses/course-route-validation.ts

```diff
@@ -458,3 +458,12 @@ export const postAttachmentValidation = {
         studentGradeInstanceId: Joi.number().optional()
     },
 };
+
+export const listAttachmentsValidation = {
+    params: {},
+    query: {
+        studentGradeId: Joi.number().optional(),
```

Comment:
You can put XOR logic here with refs if you want to get fancy.

Reply:
https://joi.dev/api/?v=17.3.0#alternatives

Reply:
I wonder how that plays with extract types, I think I would like it in both places
we'll revisit with the refactor